tests/unittests/benchmark/task/profet/test_profet_task.py

```diff
@@ -0,0 +1,375 @@
+def load_fake_data(monkeypatch, tmp_path_factory):
+    """ Fixture that prevents attempts to download the true Profet datasets, and instead generates
+    random data with the same shape.
+    """
+
+    import orion.benchmark.task.profet.profet_task
+
+    real_load_data = orion.benchmark.task.profet.profet_task.load_data
+
+    def _load_data(path: Path, benchmark: str):
+        if path == REAL_PROFET_DATA_DIR and REAL_PROFET_DATA_DIR.exists():
+            # Return real datasets.
+            logger.info(f"Testing using the real Profet datasets.")
+            return real_load_data(path, benchmark=benchmark)
+        # Generate fake datasets.
+        logger.info(f"Warning: Using random data instead of the actual profet training data.")
+        x_shape, y_shape, c_shape = shapes[benchmark]
+        X = np.random.rand(*x_shape)
+        min_y = y_min[benchmark]
+        max_y = y_max[benchmark]
+        Y = np.random.rand(*y_shape) * (max_y - min_y) + min_y
+        min_c = c_min[benchmark]
+        max_c = c_max[benchmark]
+        C = np.random.rand(*c_shape) * (max_c - min_c) + min_c
+        return X, Y, C
+
+    monkeypatch.setattr(orion.benchmark.task.profet.profet_task, "load_data", _load_data)
+    # NOTE: Need to set the item in the globals because it might already have been imported.
+    monkeypatch.setitem(globals(), "load_data", _load_data)
+
+
+@pytest.mark.timeout(10)
+@requires_profet_data
+@pytest.mark.parametrize("benchmark", ["fcnet", "forrester", "svm", "xgboost"])
+def test_download_fake_datasets(tmp_path_factory, benchmark: str, load_fake_data):
+    # TODO: Downloading the data takes a VERY long time, even though the datasets are relatively
+    # small! Would it be alright to store those datasets somewhere?
+    real_input_dir: Path = REAL_PROFET_DATA_DIR
+    real_x, real_y, real_c = load_data(real_input_dir, benchmark=benchmark)
+
+    fake_input_dir: Path = tmp_path_factory.mktemp("profet_data")
+    fake_x, fake_y, fake_c = load_data(fake_input_dir, benchmark=benchmark)
+    # assert False, (x.dtype, y.dtype, c.dtype)
+    # assert False, (x.dtype, y.dtype, c.dtype)
+    assert real_x.shape == fake_x.shape
+    assert real_y.shape == fake_y.shape
+    assert real_c.shape == fake_c.shape
+
+    assert real_x.dtype == fake_x.dtype
+    assert real_y.dtype == fake_y.dtype
+    assert real_c.dtype == fake_c.dtype
+
+    assert 0 <= real_x.min() and 0 <= fake_x.min()
+    assert real_x.max() <= 1 and fake_x.max() <= 1
+
+    min_y, max_y = y_min[benchmark], y_max[benchmark]
+    assert min_y <= real_y.min() and min_y <= fake_y.min()
+    assert real_y.max() <= max_y and fake_y.max() <= max_y
+
+    min_c, max_c = c_min[benchmark], c_max[benchmark]
+    assert min_c <= real_c.min() and min_c <= fake_c.min()
+    assert real_c.max() <= max_c and fake_c.max() <= max_c
+
+
+class ProfetTaskTests:
+    """ Base class for testing Profet tasks. """
+
+    # TODO: What would be a good set of tests for the profet tasks?
```

Comment:
I would say there is the following arguments and behaviors to test: 
### Arguments
- [ ] benchmark
- [ ] max_trials
- [ ] task_id
- [ ] seed
- [ ] input_dir
- [ ] checkpoint_dir
- [ ] train_config. This one would require to mock the model trained and some functions used to train it to verify that the training config is passed down the call stack properly. No need to test all the way down, only at end of stack is fine (deepest function called where the config is actually used). 
- [ ] device
### Behaviors
- [ ] Testing the checkpoint based on hash. No clash, unique and reproducible.
- [ ] self.name. Should it contain something else than task_id? Hash maybe?
- [x] Test seeding (test_call_is_reproducible)
- [x] Re-using checkpoint (test_instantiating_task)
- [x] configuration (test_configuration)
- [x] With gradients (test_with_gradients)

Reply:
Thanks! I will most probably not be able to add these though. I'm freaking out a little bit at the moment. Let's discuss this soon.

Reply:
Oh, there is no rush for this PR. I know you are busy with other things, it is ok! :)